src/tests/suite/daos_container.c

```diff
@@ -52,7 +54,28 @@ co_create(void **state)
 				    &info, arg->async ? &ev : NULL);
 		assert_rc_equal(rc, 0);
 		WAIT_ON_ASYNC(arg, ev);
+		assert_int_equal(info.ci_nhandles, 0);
 		print_message("container opened\n");
+
+		/* Open a second time to verify num handles was incremented */
+		rc = daos_cont_open(arg->pool.poh, str, DAOS_COO_RO, &coh2,
+				    &info, arg->async ? &ev : NULL);
+		assert_rc_equal(rc, 0);
+		WAIT_ON_ASYNC(arg, ev);
+		assert_int_equal(info.ci_nhandles, 1);
+		print_message("container opened (coh2)\n");
+
+		rc = daos_cont_close(coh2, arg->async ? &ev : NULL);
+		assert_rc_equal(rc, 0);
+		WAIT_ON_ASYNC(arg, ev);
+		print_message("container closed (coh2)\n");
+
+		/* Query with first handle (coh), verify num handles was decremented 2 -> 1 */
+		info.ci_nhandles = 0xABCD0123;
+		rc = daos_cont_query(coh, &info, NULL, arg->async ? &ev : NULL);
+		assert_rc_equal(rc, 0);
+		WAIT_ON_ASYNC(arg, ev);
+		assert_int_equal(info.ci_nhandles, 1);
```

Comment:
this one sounds correct because one handle is open.. so not sure i get it

Reply:
Here this is  a container query (as opposed to the open whose "post" execution effect would be to increment the number of open handles). So in this case query returns the number that reflects the execution of the previous open.

Reply:
will pre-increment nhandles in the next push